plugins/inlinetoolbar/plugin.js

```diff
@@ -80,25 +101,89 @@
 			};
 
 			/**
+			 * Hides the inline toolbar, detaches listeners and moves the focus back to the editable.
+			 */
+			CKEDITOR.ui.inlineToolbarView.prototype.detach = function() {
+				this._detachListeners();
+				this.hide();
+			};
+
+			/**
 			 * Places the inline toolbar next to a specified element so the tip of the toolbar's triangle
 			 * points to that element.
 			 *
 			 * @param {CKEDITOR.dom.element} element The element to which the panel is attached.
 			 */
-			CKEDITOR.ui.inlineToolbarView.prototype.create = function( element ) {
-				this.attach( element );
+			CKEDITOR.ui.inlineToolbar.prototype.create = function( element ) {
+				var output = [];
+				for ( var menuItem in this._menuItems ) {
+					this._menuItems[ menuItem ].render( this._toolbar.editor, output );
+				}
+				this._toolbar.parts.content.setHtml( output.join( '' ) );
 
-				var that = this,
-					editable = this.editor.editable();
-				this._detachListeners();
+				this._toolbar.attach( element );
+
+				var that = this._toolbar,
+					editable = this._toolbar.editor.editable();
+				this._toolbar._detachListeners();
 
-				this._listeners.push( this.editor.on( 'resize', function() {
+				this._toolbar._listeners.push( this._toolbar.editor.on( 'resize', function() {
 					that.attach( element, false );
 				} ) );
-				this._listeners.push( editable.attachListener( editable.getDocument(), 'scroll', function() {
+				this._toolbar._listeners.push( editable.attachListener( editable.getDocument(), 'scroll', function() {
 					that.attach( element, false );
 				} ) );
 			};
+
+			/**
+			 * Adds an item to the inline toolbar.
+			 *
+			 * @method
+			 * @param {String} name The menu item name.
+			 * @param {Object} element instance of CKEDITOR.ui element
+			 */
+			CKEDITOR.ui.inlineToolbar.prototype.addUIElement = function( name, element ) {
```

Comment:
Could we simplify name to `inlineToolbar.addItem`? I see that you wanted to keep compatibility with [how it's named in dialog](https://docs.ckeditor.com/ckeditor4/docs/#!/api/CKEDITOR.dialog-static-method-addUIElement), however I feel like `addItem` would be a more clear approach.

Same goes for other methods like `addItems`, `getItems`, `deleteItems`.